src/dependency-manager/src/graph/index.ts

```diff
@@ -86,4 +86,38 @@ export default abstract class DependencyGraph {
 
     return Array.from(nodes.values());
   }
+
+  removeService(service_ref: string, cleanup: boolean) {
```

Comment:
Lets make this more generic. 
rename `removeService` -> `removeNodeAndDangling` 
rename `service_ref` -> `node_ref`